### PR TITLE
added events and docs for perp lp

### PR DESCRIPTION
**Pull request by grigorevPavel**

### CODE REVIEW

File: contracts/interfaces/vaults/IPerpLPVault.sol [hunk 1]
```diff
@@ -9,51 +9,98 @@ import "../external/perp/IAccountBalance.sol";
...
+    /// @notice Returns UniswapV3 pool, which is the underlying pool with vUSDC / baseToken token pair and 0,3% fees
+    /// @return IUniswapV3Pool Uniswap pool interface, which is used in your Perp position
     function pool() external view returns (IUniswapV3Pool);
 
+    /// @notice Returns Perp Protocol Account Balance, which records most of the traders` balances (margin ratio, position size, position value)
+    /// @return IAccountBalance Perp Protocol Account Balance interface
     function accountBalance() external view returns (IAccountBalance);
 
+    /// @notice Flag, representing the status of Uni position (open / close)
+    /// @return bool Returns true is position is opened, else - false
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpLPVault.sol:51]: if

File: contracts/interfaces/vaults/IPerpLPVault.sol [hunk 2]
```diff
@@ -9,51 +9,98 @@ import "../external/perp/IAccountBalance.sol";
...
 
+    /// @notice Returns the main information about the Uni position (tick boundaries and total liquidity)
+    /// @return PositionInfo Information stored for each user's Uni position, packed into a struct
     function position() external view returns (PositionInfo memory);
 
+    /// @notice Address of the USDC used as a collateral
+    /// @return address Address of the USDC contract
     function usdc() external view returns (address);
 
+    /// @notice Returns how much your position is worth (nominated in USDC)
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpLPVault.sol:62]: This function returns 'how much pure capital you have on your account right now', not the value of your position (which may be much greater if you use leverage)

File: contracts/interfaces/vaults/IPerpLPVault.sol [hunk 3]
```diff
@@ -9,51 +9,98 @@ import "../external/perp/IAccountBalance.sol";
...
+    /// @param nft_ NFT of the vault in the VaultRegistry
+    /// @param baseToken_ Address of the virtual underlying asset that a user is trading with
+    /// @param leverageMultiplierD_ The vault capital leverage multiplier (multiplied by DENOMINATOR)
     function initialize(
         uint256 nft_,
         address baseToken_,
         uint256 leverageMultiplierD_
     ) external;
 
+    /// @notice Opens a position on UniswapV3 pool using the given tick boundaries, and to provide liquidity to the pool
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpLPVault.sol:79]: double-check all grammar everywhere please (I think a/the pool is needed)

File: contracts/vaults/PerpLPVault.sol [hunk 4]
```diff
@@ -20,21 +20,63 @@ import "../interfaces/vaults/IPerpVaultGovernance.sol";
...
     IUniswapV3Pool public pool;
+    /// @inheritdoc IPerpLPVault
     IAccountBalance public accountBalance;
 
     uint256 public constant DENOMINATOR = 10**9;
     uint256 public constant Q96 = 2**96;
 
+    /// @inheritdoc IPerpLPVault
     bool public isPositionOpened;
+    /// @notice leverageMultiplierD The vault capital leverage multiplier (multiplied by DENOMINATOR)
```
Comment (DmitryGr) [inline contracts/vaults/PerpLPVault.sol:39]: I'd explain what this multiplier does

File: contracts/interfaces/vaults/IPerpVaultGovernance.sol [hunk 5]
```diff
@@ -9,7 +9,14 @@ import "./IPerpFuturesVault.sol";
 import "./IVaultGovernance.sol";
 
 interface IPerpVaultGovernance is IVaultGovernance {
-
+    /// @notice Params that could be changed by Protocol Governance with Protocol Governance delay.
+    /// @param vault Perp Protocol internal vault contract (deposits/withdrawals)
+    /// @param clearingHouse Perp Protocol clearing house contract (open/close positions, add/remove liquidity, liquidate positions)
+    /// @param accountBalance Perp Protocol account balance contract (get position total value, add/remove base token)
+    /// @param vusdcAddress Reference to Perp Protocol vUSDC (virtual USDC after applying leverage multiplier)
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpVaultGovernance.sol:16]: what do you mean 'after applying leverage multiplier' here? The address is the address and it has nothing common with a multiplier

File: contracts/vaults/PerpLPVault.sol [hunk 6]
```diff
@@ -271,10 +313,9 @@ contract PerpLPVault is IPerpLPVault, IntegrationVault {
         );
     }
 
-    function _isStrategy(address addr) internal view returns (bool) {
-        return _vaultGovernance.internalParams().registry.getApproved(_nft) == addr;
-    }
-
+    /// @notice Close a temporarily created position
```
Comment (DmitryGr) [inline contracts/vaults/PerpLPVault.sol:316]: 'Temporarily created' -> not really true, explain clearer and more detailed

File: contracts/vaults/PerpLPVault.sol [hunk 7]
```diff
@@ -363,6 +399,39 @@ contract PerpLPVault is IPerpLPVault, IntegrationVault {
...
+    function _isStrategy(address addr) internal view returns (bool) {
+        return _vaultGovernance.internalParams().registry.getApproved(_nft) == addr;
+    }
+
+    /// @notice Calculates the position capital (nominated in USDC)
+    /// @param liquidity Uni position liquidity
+    /// @param sqrtRatioX96 A sqrt price representing the current pool prices
+    /// @param sqrtRatioAX96 A sqrt price representing the lower tick boundary
+    /// @param sqrtRatioBX96 A sqrt price representing the upper tick boundary
+    /// @return uint256 Capital of the position (nominatted in USDC)
```
Comment (DmitryGr) [inline contracts/vaults/PerpLPVault.sol:415]: add that in USDC weis

File: contracts/interfaces/vaults/IPerpLPVault.sol [hunk 8]
```diff
@@ -9,51 +9,98 @@ import "../external/perp/IAccountBalance.sol";
...
+    /// @notice Returns the main information about the Uni position (tick boundaries and total liquidity)
+    /// @return PositionInfo Information stored for each user's Uni position, packed into a struct
     function position() external view returns (PositionInfo memory);
 
+    /// @notice The address of the USDC used as a collateral
+    /// @return address The address of the USDC contract
     function usdc() external view returns (address);
 
+    /// @notice Returns how much pure capital is there on your account (nominated in USDC weis)
+    /// @return value The position capital estimated in USDC
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpLPVault.sol:63]: Not updated here

File: contracts/interfaces/vaults/IPerpVaultGovernance.sol [hunk 9]
```diff
@@ -9,7 +9,14 @@ import "./IPerpFuturesVault.sol";
 import "./IVaultGovernance.sol";
 
 interface IPerpVaultGovernance is IVaultGovernance {
-
+    /// @notice Params that could be changed by Protocol Governance with Protocol Governance delay.
+    /// @param vault Perp Protocol internal vault contract (deposits/withdrawals)
+    /// @param clearingHouse Perp Protocol clearing house contract (open/close positions, add/remove liquidity, liquidate positions)
+    /// @param accountBalance Perp Protocol account balance contract (get position total value, add/remove base token)
+    /// @param vusdcAddress Reference to Perp Protocol vUSDC (virtual USDC, it`s amount is equal to USDC amount multiplied by the current leverageMultiplier)
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpVaultGovernance.sol:16]: 'it`s amount is equal to USDC amount multiplied by the current leverageMultiplier' this is not true. Just write what is vUSDC and this is enough

File: contracts/vaults/PerpLPVault.sol [hunk 10]
```diff
@@ -271,10 +314,9 @@ contract PerpLPVault is IPerpLPVault, IntegrationVault {
         );
     }
 
-    function _isStrategy(address addr) internal view returns (bool) {
-        return _vaultGovernance.internalParams().registry.getApproved(_nft) == addr;
-    }
-
+    /// @notice Close a Uni position, if the position has non-zero funds
```
Comment (DmitryGr) [inline contracts/vaults/PerpLPVault.sol:317]: I'd explain how a permanent position emerges (see https://support.perp.com/hc/en-us/articles/5257492022809 - section Impermanent to Permanent)

File: contracts/interfaces/vaults/IPerpVaultGovernance.sol (hunk 9 above)
Comment (grigorevPavel) [reply contracts/interfaces/vaults/IPerpVaultGovernance.sol:16]: okay, I got it

File: contracts/interfaces/vaults/IPerpLPVault.sol (hunk 8 above)
Comment (grigorevPavel) [reply contracts/interfaces/vaults/IPerpLPVault.sol:63]: done

File: contracts/interfaces/vaults/IPerpFuturesVault.sol [hunk 11]
```diff
@@ -8,32 +8,65 @@ import "../external/perp/IBaseToken.sol";
...
 
+    /// @notice Returns how much pure capital is there on your account (nominated in USDC weis)
+    /// @return value The position capital estimated in USDC
     function getAccountValue() external view returns (uint256 value);
 
+    // -------------------  EXTERNAL, MUTATING  -------------------
+
+    /// @notice Updates the vault capital leverage multiplier (multiplied by DENOMINATOR)
+    /// @param newLeverageMultiplierD_ The new vault capital leverage multiplier (multiplied by DENOMINATOR)
+    /// @param isLongBaseToken_ Returns true if user`s base token position is long or not
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpFuturesVault.sol:60]: Seems a mix of two phrases, please fix

File: contracts/interfaces/vaults/IPerpFuturesVault.sol [hunk 12]
```diff
@@ -8,32 +8,65 @@ import "../external/perp/IBaseToken.sol";
...
+    /// @param deadline The restriction on when the transaction should be executed, otherwise, it fails
     function updateLeverage(uint256 newLeverageMultiplierD_, bool isLongBaseToken_, uint256 deadline) external;
 
+    /// @notice Adjusts the current position to the multiplied capital multiplied by the current leverage multiplier. (capital nominated in USDC weis)
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpFuturesVault.sol:64]: 'multiplied' twice

File: contracts/interfaces/vaults/IPerpFuturesVault.sol [hunk 13]
```diff
@@ -8,32 +8,65 @@ import "../external/perp/IBaseToken.sol";
...
+    /// @param deadline The restriction on when the transaction should be executed, otherwise, it fails
     function adjustPosition(uint256 deadline) external;
 
+    /// @notice Closes a position on the UniswapV3 pool and remove all the provided liquidity
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpFuturesVault.sol:68]: We don't 'provide liquidity', rather take long or shorts

File: contracts/interfaces/vaults/IPerpFuturesVault.sol [hunk 14]
```diff
@@ -8,32 +8,65 @@ import "../external/perp/IBaseToken.sol";
...
+    /// @dev If a position has not been opened, it reverts
```
Comment (DmitryGr) [inline contracts/interfaces/vaults/IPerpFuturesVault.sol:69]: That's not true

**Review by DmitryGr — APPROVED** [review]

**Pull request merged**